src/OptimalTransport.jl

```diff
@@ -211,21 +230,26 @@ where ``H`` is the entropic regulariser, ``H(\\gamma) = -\\sum_{i, j} \\gamma_{i
 
 A pre-computed optimal transport `plan` may be provided.
 
+Note that for a common cost `C`, multiple histograms may be provided for a batch computation by passing `mu` and `nu`
+as matrices whose columns `mu[:, i]` and `nu[:, i]` correspond to pairs of histograms. 
+The output in this case is a vector `c` of coupling matrices such that `c[i]` is the transport cost for `mu[:, i]` and `nu[:, i]`.
+
 See also: [`sinkhorn`](@ref)
 """
 function sinkhorn2(μ, ν, C, ε; plan=nothing, kwargs...)
     γ = if plan === nothing
         sinkhorn(μ, ν, C, ε; kwargs...)
     else
         # check dimensions
-        size(C) == (length(μ), length(ν)) ||
-            error("cost matrix `C` must be of size `(length(μ), length(ν))`")
+        size(C) == (size(μ, 1), size(ν, 1)) || error(
+            "cost matrix `C` must be of size `(size(μ, dims = 1), size(ν, dims = 1))`",
+        )
         size(plan) == size(C) || error(
             "optimal transport plan `plan` and cost matrix `C` must be of the same size",
         )
```

Comment:
This should not be enforced anymore, better
```suggestion
        (size(plan, 1), size(plan, 2)) == size(C) || error(
            "optimal transport plan `plan` and cost matrix `C` must be of the same size",
        )
```